src/main/app/src/components/ScheduleActivityWrapper.js

```diff
@@ -0,0 +1,15 @@
+import React from 'react';
+import ScheduleActivityPage from '../pages/ScheduleActivityPage';
+
+/** This is a wrapper component for the schedule activity page.
+    It holds the state of the random activities generated to prevent
+    unnecessary rerenderes. */
+const ScheduleActivityWrapper = props => {
+  // Todo: Move randomActivities state from the schedule activity page to here.
+  const [randomActivities, updateRandomActivities] = React.useState([]);
+  return (
+    <ScheduleActivityPage {...props} />
+  )
```

Comment:
This is exactly what we needed! Perfect. Did you test it thoroughly?

Reply:
Yes! I will go ahead and refactor our code with this so it can look a lot cleaner.